java/rocksjni/options.cc

```diff
@@ -3621,6 +3635,19 @@ void Java_org_rocksdb_ColumnFamilyOptions_setTableFactory(
           reinterpret_cast<ROCKSDB_NAMESPACE::TableFactory*>(jfactory_handle));
 }
 
+/*
+ * Method:    setSstPartitionerFactory
+ * Signature: (JJ)V
+ */
+void Java_org_rocksdb_ColumnFamilyOptions_setSstPartitionerFactory(
+    JNIEnv*, jobject, jlong jhandle, jlong factory_handle) {
+  auto* options =
+      reinterpret_cast<ROCKSDB_NAMESPACE::ColumnFamilyOptions*>(jhandle);
+  auto* factory = reinterpret_cast<ROCKSDB_NAMESPACE::SstPartitionerFactory*>(
```

Comment:
I think this should be a std::shared_ptr<...> for consistency with how the other Java code behaves.  And this will eliminate the need for the Clone method as well.

Reply:
It is now as shared_ptr only.